ReactWindows/ReactNative/Chakra/Executor/NativeJavaScriptExecutor.cs

```diff
@@ -214,7 +214,7 @@ public void SetCallSyncHook(Func<int, int, JArray, JToken> callSyncHook)
                 throw new ArgumentNullException(nameof(callSyncHook));
 
             _executor.SetCallSyncHook((moduleId, methodId, args) =>
-                callSyncHook(moduleId, methodId, JArray.Parse(args)).ToString(Formatting.None));
+                callSyncHook(moduleId, methodId, JArray.Parse(args))?.ToString(Formatting.None) ?? "");
```

Comment:
wouldn't "null" be a better return value since it would result in a valid TokenType for native sync methods that return strings, arrays, and objects? I think "" will result in a JTokenType of string, which may cause other weird JS errors on shutdown.

Reply:
I could try, the hook itself expected a string as output (see that ToString), so I was afraid a null would break things "more". I'll give it a try.

---
In reply to: [192269360](https://github.com/Microsoft/react-native-windows/pull/1847#discussion_r192269360) [](ancestors = 192269360)

Reply:
Nope, doesn't work, crashes with a stack like this:

  KernelBase.dll!RaiseException(unsigned long dwExceptionCode, unsigned long dwExceptionFlags, unsigned long nNumberOfArguments, const unsigned __int64 * lpArguments) Line 922 C
  vcruntime140d_app.dll!_CxxThrowException(void * pExceptionObject, const _s__ThrowInfo * pThrowInfo) Line 136 C++
  vccorlib140d_app.dll!__abi_WinRTraiseNullReferenceException() Line 493 C++
> ChakraBridge.dll!__abi_WinRTraiseException(long __hrArg) Line 1101 C++
  ChakraBridge.dll!__abi_ThrowIfFailed(long __hrArg) Line 102 C++
  ChakraBridge.dll!ChakraBridge::CallSyncHandler::__abi_IDelegate::Invoke(int moduleId, int methodId, Platform::String ^ args) C++
  ChakraBridge.dll!ChakraBridge::CallSyncHandler::NonVInvoke(int moduleId, int methodId, Platform::String ^ args) C++
  ChakraBridge.dll!NativeCallSyncHook(void * callee, bool isConstructCall, void * * arguments, unsigned short argumentCount, void * callbackState) Line 273 C++

I'll stick to "".

---
In reply to: [192270098](https://github.com/Microsoft/react-native-windows/pull/1847#discussion_r192270098) [](ancestors = 192270098,192269360)

Reply:
I agree with Matt - I think we should return the null value and force ChakraBridge to handle the null case (perhaps by returning Undefined to the JavaScript function). Seems to me returning empty string will just cause a JavaScript error when ChakraBridge calls `JSON.parse`

Reply:
I suspect that can be done by bypassing that conversion to string/etc. On the other hand I'm really not sure how JSON.parse("") would fare better than JSON.parse(null).
In other words we make the bridge smoother for a caller that doesn't expect what we return anyway...

Parking this for now, not having too much time, I may look at it later

---
In reply to: [192273283](https://github.com/Microsoft/react-native-windows/pull/1847#discussion_r192273283) [](ancestors = 192273283)

Reply:
I think you misunderstood me/the Newtonsoft docs -- literally `.Parse("null")` is what I was recommending. Sorry for not being clearer.

Reply:
Yeah, I clearly misunderstood. I'll test it when I have a chance.

---
In reply to: [192865585](https://github.com/Microsoft/react-native-windows/pull/1847#discussion_r192865585) [](ancestors = 192865585)